core/src/nucleus/mod.rs

```diff
@@ -182,24 +183,34 @@ impl FunctionResult {
 #[allow(large_enum_variant)]
 pub enum Action {
     InitApplication(Dna),
-    ReturnInitializationResult(bool),
+    ReturnInitializationResult(Option<String>),
     ExecuteZomeFunction(FunctionCall),
     ReturnZomeFunctionResult(FunctionResult),
     ValidateEntry(EntrySubmission),
 }
 
 /// Reduce ReturnInitializationResult Action
 /// On initialization success, set Initialized status
-/// otherwise reset the nucleus
-fn reduce_rir(nucleus_state: &mut NucleusState, has_succeeded: bool) {
-    assert!(nucleus_state.status == NucleusStatus::Initializing);
-    if has_succeeded {
-        (*nucleus_state).status = NucleusStatus::Initialized
-    } else {
-        *nucleus_state = NucleusState::new();
+/// otherwise set the failed message
+fn reduce_rir(nucleus_state: &mut NucleusState, result: &Option<String>) {
+    //assert!(nucleus_state.status == NucleusStatus::Initializing);
+    match result {
+        None => (*nucleus_state).status = NucleusStatus::Initialized,
+        Some(err) => (*nucleus_state).status = NucleusStatus::InitializationFailed(err.clone()),
     };
 }
 
+fn return_initialization_result(
+    result: Option<String>,
```

Comment:
i think (haven't tried) that if this was `Option<&str>` you wouldn't need to `.to_string()` as much below